Microsoft.Azure.Cosmos/src/Telemetry/ClientTelemetry.cs

```diff
@@ -209,6 +200,33 @@ await this.processor
             DefaultTrace.TraceInformation("Telemetry Job Stopped.");
         }
 
+        /// <summary>
+        /// This Task makes sure, processing task is timing out after 5 minute of timeout
+        /// </summary>
+        /// <param name="telemetryDate"></param>
+        /// <param name="processingTask"></param>
+        /// <param name="timeout"></param>
+        internal static async Task RunProcessorTaskAsync(string telemetryDate, Task processingTask, TimeSpan timeout)
+        {
+            using (CancellationTokenSource tokenForDelayTask = new CancellationTokenSource())
+            {
+                Task delayTask = Task.Delay(timeout, tokenForDelayTask.Token);
+                
+                Task resultTask = await Task.WhenAny(processingTask, delayTask);
+                if (resultTask == delayTask)
+                {
+                    DefaultTrace.TraceVerbose($"Processor task with date as {telemetryDate} is canceled as it did not finish in {ClientTelemetryOptions.ClientTelemetryProcessorTimeOut}");
```

Comment:
just realized the method takes the timespan as parameter. If we want to use `ClientTelemetryOptions.ClientTelemetryProcessorTimeOut` on the error trace then we need to use it on the Task.Delay and remove it from the parameters, or, change the message to:

```suggestion
                    DefaultTrace.TraceVerbose($"Processor task with date as {telemetryDate} is canceled as it did not finish in {timeout}");
```